api/v1beta1/types.go

```diff
@@ -799,6 +809,14 @@ type AzureBastion struct {
 	Subnet SubnetSpec `json:"subnet,omitempty"`
 	// +optional
 	PublicIP PublicIPSpec `json:"publicIP,omitempty"`
+	// BastionHostSkuName configures the tier of the Azure Bastion Host. Can be either Basic or Standard. Defaults to Basic.
```

Comment:
Ah I definitely missed this. Would it be more appropriate to default via kubebuilder, or the defaulting webhook?

Reply:
I'm leaning towards webhook since it would be persisted that way

Reply:
Since the [Azure API defaults to Standard](https://learn.microsoft.com/en-us/rest/api/virtualnetwork/bastion-hosts/create-or-update?tabs=HTTP#sku), is there a reason we're setting the default in CAPZ to Basic?

Reply:
I guess the only reason could be that `Standard` [is more expensive](https://azure.microsoft.com/en-us/pricing/details/azure-bastion/) but IMO we should default to whatever Azure defaults to.

+1 for persisting, kubebuilder should persist and is good enough for simple defaulting where no conditions are involved

Reply:
Whenever I create a bastion via the portal, it seems like Basic was the default, but if the API defaults to Standard, we can make that change. The only thing is that Bastions were previously defaulted to Basic on CAPZ.

Reply:
I guess I would defer to the actual behavior then and have CAPZ default it to Basic.

Reply:
Actually I think I was mistaken. I just double-checked in the portal and it seems to default to Standard! That's my bad.

Reply:
I don't think you were actually. I just spun up the private cluster template from the main branch and I see the Bastion has the "Basic" SKU, but using `az network bastion create` creates a "Standard" one by default. I couldn't set the `DNSName` or `PrivateIPAllocationMethod` parameters from the CLI like CAPZ does though, so I wonder if those affect the documented default?

Reply:
Actually, setting this up using the same parameters CAPZ passes using
```
az resource create -g private-11449 -n private-11449-azure-bastion --resource-type Microsoft.Network/bastionHosts --is-full-object --properties '@Desktop/bastion.json' --debug --verbose
```
to send this JSON:
```json
{
  "location": "eastus",
  "properties": {
    "dnsName": "private-11449-azure-bastion-bastion.bastion.azure.com",
    "ipConfigurations": [
      {
        "name": "private-11449-azure-bastion-bastionIP",
        "type": "Microsoft.Network/bastionHosts/bastionHostIpConfigurations",
        "properties": {
          "privateIPAllocationMethod": "Dynamic",
          "publicIPAddress": {
            "id": "/subscriptions/XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX/resourceGroups/private-11449/providers/Microsoft.Network/publicIPAddresses/private-11449-azure-bastion-pip"
          },
          "subnet": {
            "id": "/subscriptions/XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX/resourceGroups/private-11449/providers/Microsoft.Network/virtualNetworks/private-11449-vnet/subnets/AzureBastionSubnet"
          }
        }
      }
    ]
  }
}
```

does result in a "Basic" SKU Bastion. Still not exactly sure why it's not "Standard" though.

The only differences I can see from the previous CLI command are the `dnsName` (defaulted vs. `name+"-bastion"` and `ipConfigurations[0].name` (defaulted vs. `name+"-bastionIP"`. I'm working through to see exactly what the cause is.

(The CLI uses the 2022-01-01 API version whereas CAPZ uses 2021-08-01, but even with just the CLI and I can send two different Bastions and get two different defaults for SKU.)

Reply:
That's actually really interesting.. thanks for testing this! Looks like the default SKU isn't exactly standardized, so we might need to decide for ourselves what SKU to default to. The Basic SKU is cheaper and can be upgraded to Standard, and CAPZ used to default to Basic without a way to change it. However, Standard comes with a lot more features such as native client support. @CecileRobertMichon do you have an insight to what SKU we should default to?

Reply:
I would prefer to default to the basic SKU due to cost reasons. When someone needs more involved feature, they know what they are asking for and can opt to upgrade. Also, for more users basic SKU will suffice.

Reply:
I received confirmation from the Bastion devs that the Standard SKU should be the default. However, I think cost can still be a major concern here, especially since CAPZ previously defaulted to the Basic tier. I think I'm leaning towards defaulting to Basic since Standard is only really needed if the user chooses to set `enableTunneling` to true

Reply:
As long as we persist the defaulted field, we can always switch the default later without impacting existing users.